### PR TITLE
HL 83 - Fix settle.ts script

### PR DESCRIPTION
**Pull request by deBhal**
[Link to ticket](https://dltx.atlassian.net/browse/HL-83)

Firstly this PR unbreaks the RPC Provider (connection to the chain) by ensuring that the `.env` file is loaded before the RPC provider is instantiated.

Secondly, we handle exceptions, allowing the script to continue past a failure interacting with a contract.

It also incrementally improves the logic by settling the newest bets first, only fetching unsettled bets, and limiting by the number of (unsettled) bets while loosening the time limitation. This should avoid the situation where the script finds no bets and also stop it from spamming the api as hard.

Finally, the script's output is much improved, adding the reason provided from RPC failures. e.g. this is output from testing on utils.horse.link:
```
adding result for market 0x30313934303557414730320000000000 0x7b4e16cc54f3b7bc3fd362e47d7b22860cfc31441afcfdf65ce7cc9995ecb7eb
adding result for market 0x30313934303557414730320000000000 setResult FAILED execution reverted: setResult: Result already set
settle bet BET_0x39a93a77cff7c3d5e3233cff9b9868172a2c6615_449(449), receipt FAILED: execution reverted: _payout: Payout date not reached
settle bet BET_0x39a93a77cff7c3d5e3233cff9b9868172a2c6615_448(448), receipt FAILED: execution reverted: _payout: Payout date not reached
settle bet BET_0x39a93a77cff7c3d5e3233cff9b9868172a2c6615_441(441), receipt FAILED: execution reverted: _payout: Payout date not reached
settle bet BET_0x39a93a77cff7c3d5e3233cff9b9868172a2c6615_440(440), receipt 0xefa5e935f56a790c2411a4a07f86cb40f907ea4dd8ae9697f8ee59e74c50272f
settle bet BET_0x39a93a77cff7c3d5e3233cff9b9868172a2c6615_439(439), receipt 0xf9a421df2953d23b7354b61cc4cc4d65814c3a19b68abdcb76e6c43134d0a869
settle bet BET_0x39a93a77cff7c3d5e3233cff9b9868172a2c6615_438(438), receipt 0x72ebe93d35e48b07100bd3dde24e4341e68bdc08e7cb26c6ba97ecfcb23f0501
settle bet BET_0x39a93a77cff7c3d5e3233cff9b9868172a2c6615_437(437), receipt 0x974d08ba3af14efae1fe6c74c9d5af0fa8290d883d9a86ac34e56b5a3bef0155
settle bet BET_0x39a93a77cff7c3d5e3233cff9b9868172a2c6615_436(436), receipt 0x16182e04c8450b290f63dd82f295f545a3215d1abb7fb03a69c362b5ce2fa3f8
request failure for market 019405QRI08: {
  responseStatus: 404,
  responseStatusText: 'Not Found',
  url: 'https://alpha.horse.link/api/markets/result/019405QRI08?sign=true',
  data: undefined
}
```

### CODE REVIEW

**Review by bitcoinbrisbane — DISMISSED** [review]

File: scripts/settle.ts [hunk 1]
```diff
@@ -48,10 +45,16 @@ export async function main() {
 	console.log(`Current Time: ${now}`);
 
 	// Now less 2 hours
```
Comment (pondpiu) [inline scripts/settle.ts:47]: misleading comment?

File: scripts/settle.ts [hunk 2]
```diff
@@ -48,10 +45,16 @@ export async function main() {
...
 
-	const bets: BetDetails[] = await getSubgraphBetsSince(closeTime);
+	const bets: BetDetails[] = await getSubgraphBetsSince(closeTime, {
+		unsettledOnly: true,
+		maxResults: 150
+	});
+
+	console.log(`Found ${bets.length} unsettled bets`);
 
 	// Process up to 50 most recent, starting with most recent
```
Comment (pondpiu) [inline scripts/settle.ts:59]: misleading comment?

File: scripts/utils.ts [hunk 3]
```diff
@@ -179,13 +183,30 @@ export const getGasPriceFromEnv = (): ethers.BigNumber => {
...
 	const timeString = Math.floor(createdAtGt);
+
+	const { unsettledOnly = false, maxResults = 1000, payoutAtLt } = options;
+
+	const whereClauses = [
+		`createdAt_gt: "${timeString}"`,
+		unsettledOnly ? ", settled_not: true" : "",
+		payoutAtLt ? `payoutAt_lt: ${payoutAtLt}` : ""
+	];
+	const whereClause = whereClauses.filter(Boolean).join(",");
```
Comment (brettvniekerk) [inline scripts/utils.ts:205]: why filter `Boolean` ? `whereClause.join(",")` should produce the same result

Comment (brettvniekerk) [reply scripts/utils.ts:205]: ah nvm, i see the ternaries in there

**Review by brettvniekerk — APPROVED** [review]

File: scripts/utils.ts (hunk 3 above)
Comment (deBhal) [reply scripts/utils.ts:205]: It cleans up the empty strings and avoids extra commas:

`['foo','',false, null, 'bar'].join(',')` // `'foo,,false,,bar'`
vs
`['foo','',false, null, 'bar'].filter(Boolean).join(',')` // `'foo,bar'`

Comment (deBhal) [reply scripts/utils.ts:205]: :TIL: Looking at that output, it looks like we could use nulls in the ternaries instead. I wonder if I prefer that

**Review by bitcoinbrisbane — APPROVED** [review]

**Pull request merged**